tempesta_fw/str.h

```diff
@@ -99,27 +99,32 @@
  * 		  string and the least significant byte for flags;
  */
 typedef struct {
-	void		*ptr;
 	struct sk_buff	*skb;
 	unsigned long	len;
 	unsigned char	eolen;
-	unsigned int	flags;
+
+	unsigned int	chunknum : 24;
+	unsigned int	flags : 8;
```

Comment:
Callers of `__str_grow_tree()` and the function itself use chunks number as a part of flags, and that looks easy and good. So it seems it has sense to define the fields as (please note `,` instead of `;`)

        union {
                unsigned int    chunknum: 24,
                                        flags: 8;
                unsigned int __flags;
        }